dymos/load_case.py

```diff
@@ -51,83 +78,101 @@ def load_case(problem, previous_solution):
                                               nodes='control_input', kind='slinear'),
                             units=prev_control_units)
 
         # Set the output polynomial control outputs from the previous solution as the value
-        for polynomial_control_name, options in phase.polynomial_control_options.items():
-            polynomial_control_path = [s for s in phase_outputs if
-                                       s.endswith(f'{phase_name}.polynomial_controls:{polynomial_control_name}')][0]
-            prev_polynomial_control_path = [s for s in prev_outputs if
-                                            s.endswith(f'{phase_name}.'
-                                                       f'polynomial_controls:{polynomial_control_name}')][0]
-            prev_polynomial_control_val = prev_outputs[prev_polynomial_control_path]['value']
-            prev_polynomial_control_units = prev_outputs[prev_polynomial_control_path]['units']
+        for pc_name, options in phase.polynomial_control_options.items():
+            polynomial_control_path = [s for s in phase_vars if
+                                       s.endswith(f'{phase_name}.polynomial_controls:{pc_name}')][0]
+            prev_polynomial_control_path = [s for s in prev_vars if
+                                            s.endswith(f'{phase_name}.polynomial_controls:{pc_name}')][0]
+            prev_polynomial_control_val = prev_vars[prev_polynomial_control_path]['value']
+            prev_polynomial_control_units = prev_vars[prev_polynomial_control_path]['units']
             problem.set_val(polynomial_control_path, prev_polynomial_control_val,
                             units=prev_polynomial_control_units)
 
         # Set the timeseries parameter outputs from the previous solution as the parameter value
-        for parameter_name, options in phase.parameter_options.items():
-            parameter_path = [s for s in phase_inputs if s.endswith(f'{phase_name}.parameters:{parameter_name}')][0]
-            prev_parameter_path = [s for s in prev_outputs if
-                                   s.endswith(f'{phase_name}.timeseries.parameters:{parameter_name}')][0]
-            prev_parameter_val = prev_outputs[prev_parameter_path]['value'][0, ...]
-            prev_parameter_units = prev_outputs[prev_parameter_path]['units']
-
-            problem.set_val(parameter_path, prev_parameter_val, units=prev_parameter_units)
+        for param_name, options in phase.parameter_options.items():
+            prev_match = [s for s in prev_vars if s.endswith(f'{phase_name}.parameters:{param_name}')]
+            if prev_match:
+                # In previous outputs
+                prev_data = prev_vars[prev_match[0]]
+                prev_param_val = prev_data['value']
+                prev_param_units = prev_data['units']
+                param_path = [s for s in phase_vars if s.endswith(f'{phase_name}.parameters:{param_name}')][0]
```

Comment:
I think you can do `param_path = prev_match[0]` unless there is something I am missing.

Reply:
There's the possibility that the same phase name is now wrapped in a different hierarchy (maybe a different trajectory name, for instance) and I wanted to allow for that.

Reply:
Thinking further about this, the current load_case capability allows for phases that aren't encapsulated in a trajectory object.  But if the user has two identical trajectory structures with the same enclosed phase names in their model, it will break this implementation.  I think for version 1.0 we should be more strict about how we grab phase variable names - The previous path must match the current path, and perhaps we eventually allow some sort of remapping via arguments (e.g. phase at 'trajA.phaseA' in the database should map to 'trajB.phaseC' in the current case being loaded).

Reply:
Oh, I see the difference now. I should have just done a block search.